### DISCUSSION TITLE
[Question] this is good to create modular os and use that for each service?

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [dotnet/runtime#103474] [Q&A]
I dont know why did this occur in my mind, but i think this must be good to create modular os that is for backend project each project can add some modules to os that need and each service have thats own os and services installed and run in os without any third party libraries and projects 
If I got my view right, i told we have os that os just run our application and that os is our application 😅
For example Steps to execute asp.net core:
1. Run .net runtime as os
2. Running application 
3. Each problem is called parent os

**Comment by Clockwork-Muse** [comment]
You're going to have a much easier time just using a containerized version of your application (which isn't exactly what you're thinking of, but for most practical purposes currently will be).  [This is a one-command option](https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container?pivots=dotnet-8-0) (assuming some minor project tweaks) - `dotnet publish --os linux --arch x64 /t:PublishContainer`

This _does_ already exist generally, but is more complicated to set up and work with.  This ends up either being some form of [unikernel](https://en.wikipedia.org/wiki/Unikernel) (although I'm not sure how well C# supports this), or building a custom OS image, for example with [Yocto](https://www.yoctoproject.org/) (often intended for small embedded devices).
